state/sshhostkeys.go

```diff
@@ -45,26 +46,57 @@ func (st *State) GetSSHHostKeys(tag names.MachineTag) (SSHHostKeys, error) {
 	return SSHHostKeys(doc.Keys), nil
 }
 
+// keysEqual checks if the ssh host keys are the same between two sets.
+// we shouldn't care about the order of the keys.
+func keysEqual(a, b []string) bool {
+	if len(a) != len(b) {
+		return false
+	}
+	for i := range a {
```

Comment:
Based on the function docs shouldn't you be sorting the lists first or using a map to implement exists-in-set semantics?

Reply:
we need them to be bi-directionally equal, I was just being lazy by not sorting. patch is updated.